### PR TITLE
added scrape-meta and scrape for riverside pd

### PR DESCRIPTION
**Pull request by naumansharifwork**
Hey fellows,
Created scraper for City of Riverside,
Below is the tree screenshot and meta json
i have made folder name as case numbers.
image
[ca_river_side_pd.json](https://github.com/user-attachments/files/16470468/ca_river_side_pd.json)

### CODE REVIEW

**Comment by naumansharifwork** [comment]
@zstumgoren @newsroomdev please review this PR as well i have formatted the json as it was mentioned in the discussion
[ca_river_side_pd.json](https://github.com/user-attachments/files/16480680/ca_river_side_pd.json)

**Review by tarakc02 — CHANGES_REQUESTED** [review]
@naumansharifwork, sorry i misesed this earlier, but filename/slug should be "riverside_pd" rather than "river_side_pd". otherwise i don't have any other changes, it looks good to me.

**Review by tarakc02 — CHANGES_REQUESTED** [review]

File: clean/ca/riverside_pd.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        for asset in metadata:
+            print("downloading...: ", asset["name"])
+            url = asset["asset_url"]
+            dl_path = self._make_download_path(asset)
+            time.sleep(throttle)
+            dl_assets.append(self.cache.download(str(dl_path), url))
+        return dl_assets
+
+    def _make_download_path(self, asset):
+        folder_name = asset["case_number"]
```
[the remark this answers is not in the document]
Comment (naumansharifwork) [reply clean/ca/riverside_pd.py]: @tarakc02 fixed

**Review by tarakc02 — APPROVED** [review]
looks good to me!

**Comment by naumansharifwork** [comment]
Hey @zstumgoren please review this as well, Thanks

File: clean/ca/riverside_pd.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                        if (
+                            self.sec_website_url not in link_href
+                            and self.un_sec_website_url not in link_href
+                        ):
+                            link_href = f"{self.sec_website_url}{link_href}"
+                        name = link_href.split("/")[-1]
+                        name = urllib.parse.unquote(name)
+                        payload = {
+                            "asset_url": link_href,
+                            "case_num": case_id,
```
Comment (newsroomdev) [inline clean/ca/riverside_pd.py:80]: small tweak to match spec
```suggestion
                            "case_id": case_id,
```

File: clean/ca/riverside_pd.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                            "parent_page": str(filename),
+                            "details": {"date": date},
+                        }
+                        metadata.append(payload)
+                    time.sleep(throttle)
+        outfile = self.data_dir.joinpath(f"{self.agency_slug}.json")
+        self.cache.write_json(outfile, metadata)
+        return outfile
+
+    def scrape(self, throttle: int = 4, filter: str = "") -> List[Path]:
```
Comment (newsroomdev) [inline clean/ca/riverside_pd.py:92]: note: this will be deprecated in #82

**Review by newsroomdev — CHANGES_REQUESTED** [review]

**Comment by newsroomdev** [comment]
This is looking very close to being done! Nice work. Two changes and this should be good to merge. Let me know when you'd like me to take another look, and thank you for your diligence as this codebase adds downstream dependencies

**Comment by naumansharifwork** [comment]
Hey @newsroomdev Thanks for the review, the requested changes are done please check Thanks.

File: clean/ca/riverside_pd.py (hunk 3 above)
Comment (newsroomdev) [reply clean/ca/riverside_pd.py]: removed 👍

File: clean/ca/riverside_pd.py (hunk 2 above)
Comment (newsroomdev) [reply clean/ca/riverside_pd.py]: fixed 👍

**Review by newsroomdev — APPROVED** [review]
This is looking good! Feel free to merge it in when you're ready

**Pull request merged**